src/test/scala/com/databricks/spark/redshift/RedshiftSuite.scala

```diff
@@ -0,0 +1,458 @@
+/*
+ * Copyright 2015 TouchType Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.databricks.spark.redshift
+
+import java.io.File
+import java.sql.Connection
+import java.util.Properties
+
+import org.apache.spark.SparkContext
+import org.apache.spark.sql.jdbc.DefaultJDBCWrapper
+import org.apache.spark.sql.{Row, SQLContext, SaveMode}
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.{BeforeAndAfterAll, FunSuite, Matchers}
+
+/**
+ * Tests main DataFrame loading and writing functionality
+ */
+class RedshiftSuite extends FunSuite with Matchers with MockFactory with BeforeAndAfterAll {
+  val jdbcUrl = s"${System.getenv("aws_redshift_jdbc_url")}?" +
```

Comment:
This is messy / hard-to-read. This could be simplified by just using the `vals` defined two lines down.

Also, I think it's generally conventional to put environment variables in all-caps.